### ISSUE TITLE
ncu --deep override and ignore parent .ncurc.yml settings (reject)

### ISSUE THREAD

**Issue by art-ws** [raineorshine/npm-check-updates#895]
- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [x] I am using the latest version of `npm-check-updates`
- [x] I am using `node >= 10.17`

---------------------------
## Abstract

If we have a workspace (like [rushjs monorepo](https://rushjs.io/)) with hierarchy of nested projects sometimes we need to reject some specific packages for all projects in workspace. Currently it's possible to achieve via file system config with package-specific `.ncurc.yml` which works great after the fix #826 , but it require a lot of identical file updates in existing config files and explicit creating of additional configs. If we define a parent `.ncurc.yml` it works as expected only for packages without package-specific `.ncurc.yml` config.

## Steps to Reproduce

1. Define `.ncurc.yml` for some projects in root folder and reject some package
2. Define `.ncurc.yml`for some nested projects without rejected package
3. Not define `.ncurc.yml`for some nested package
3. Run `ncu -u --deep` 

## Current Behavior

`ncu` correctly recognize parent `.ncurc.yml` and apply it correctly only for nested packages, which have no own settings. If some project contains own local `.ncurc.yml`, then parent settings are discarded. 
As result different packages in monorepo even on same level of hierarchy may contains different versions and whole workspace state became inconsistent.

## Expected Behavior

Would be nice to inherit settings from parent configs (at least for 'reject' option), so tool behevior will be more consistent.

**Comment by art-ws** [comment]
We tested locally a fix with literally few lines of code and it works as expected.
If you approve we can prepare the PR.

File `lib/index.js`

```javascript
...
async function getAnalysis() {
...
       const packages = await previousPromise
        // copy object to prevent share .ncurc options between different packageFile, to prevent unpredictable behavior
        const rcResult = getNcurc({ packageFile })

       // BEGIN: Possible #895 fix
        if (options.reject && rcResult && rcResult.config && rcResult.config.reject) {
          rcResult.config.reject = [...new Set([...options.reject, ...rcResult.config.reject])]
        }
       // END

        const pkgOptions = {
          ...options,
          ...rcResult && rcResult.config,
          packageFile,
        }
        const [pkgData, pkgFile] = await findPackage(pkgOptions)
...
```

Currently problem is in next lines which fully overrides the nested object property values, but, probably kind of more smart object merge implementation will be better here. WDT?

**Comment by raineorshine** [comment]
Thanks for the suggestion and thanks for digging into some code! 

I agree it would be useful to be able to specify a recursive option. We don't want to make the config recursive by default as we have to assume that separate packages with separate package.json files have their own dependencies and their own rules, even if they are within the same repo. An opt-in option `recursive: true` would be a reasonable way to make this behavior available. 

It is still something I would only recommend to people with caution, as it breaks an invariant: That running `ncu` from the root package folder has the same behavior as running `ncu` from the sub-package folder. If you use this recursive feature then running `ncu` from the sub-package folder would not know about the config higher up. In general I recommend that people use separate configs, especially if they only have a few sub-packages.

**Comment by art-ws** [comment]
Fix in `v11.6.0`
